include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -287,10 +292,54 @@ auto computePanelReflectors(MatrixLike& mat_a, const matrix::SubPanelView& panel
     panel_tiles.emplace_back(matrix::splitTile(mat_a(i), spec));
   }
 
-  return ex::when_all_vector(std::move(panel_tiles)) |
-         dlaf::internal::transform(dlaf::internal::Policy<Backend::MC>(
-                                       pika::execution::thread_priority::high),
-                                   std::move(panel_task)) |
+  const size_t nthreads = getReductionToBandPanelNWorkers();
+  return ex::when_all(ex::just(std::make_shared<pika::barrier<>>(nthreads)),
+                      ex::just(std::vector<common::internal::vector<T>>{}),  // w (interally required)
+                      ex::just(common::internal::vector<T>{}),               // taus
+                      ex::when_all_vector(std::move(panel_tiles))) |
+         ex::let_value([nthreads, nrefls, cols = panel_view.cols()](auto& barrier_ptr, auto& w,
+                                                                    auto& taus, auto& tiles) {
+           return ex::schedule(dlaf::internal::getBackendScheduler<Backend::MC>(
+                      pika::execution::thread_priority::high)) |
+                  ex::bulk(nthreads,
+                           [=, &barrier_ptr, &taus, &w, &tiles](const size_t index) {
```

Comment:
This is a note for a potential simplification which can be done in a separate PR as well. I'm realizing now that the outer `let_value` may not be necessary because `bulk` has a similar capability.

Instead of
```
when_all(...) |
let_value([](...) {
    return schedule() | bulk([](...) { ... });
});
```
I think this can be written as
```
when_all(...) |
transfer() |
bulk([](int i, t1& x1, t2& x2, ...) { ... }) |
then(drop_whatever_you_dont_need_from_when_all); // bulk will send x1, x2 etc. by value
```
See https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2300r6.html#example-async-inclusive-scan for a similar example.

Changing this is not a blocker for this PR.